src/changeThreadUserNickname.js

```diff
@@ -0,0 +1,36 @@
+"use strict";
+
+var utils = require("../utils");
+var log = require("npmlog");
+
+module.exports = function(defaultFuncs, api, ctx) {
+  return function changeThreadUserNickname(nickname, threadID, participantID, callback) {
+
+    var form = {
+      'nickname': nickname,
+      'participant_id': participantID,
+      'thread_or_other_fbid': threadID
+    };
+
+    defaultFuncs
+      .post("https://www.messenger.com/messaging/save_thread_nickname/?source=thread_settings&dpr=1", ctx.jar, form)
+      .then(utils.parseAndCheckLogin(ctx.jar, defaultFuncs))
+      .then(function(resData) {
+        if (resData.error === 1545014) {
+          throw {error: "Trying to change user nickname wasnt in thread"};
```

Comment:
I don't understand this error message.  Maybe you mean "Trying to change the nickname of a user not in this thread" or something similar?